frontend/packages/dev-console/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -49,17 +79,35 @@ export const createPipelineForImportFlow = async (formData: GitImportFormData) =
         return { ...param, default: git.dir.replace(/^\//, '') || param.default };
       case 'IMAGE_NAME':
         return { ...param, default: getImageUrl(name, namespace) };
+      case 'DOCKERFILE':
+        return { ...param, default: dockerfilePath };
       default:
         return param;
     }
   });
 
-  if (template.spec.resources?.find((r) => r.type === 'git' && r.name === 'app-source')) {
-    await createGitResource(git.url, namespace, git.ref);
-  }
-  if (template.spec.resources?.find((r) => r.type === 'image' && r.name === 'app-image')) {
-    await createImageResource(name, namespace);
-  }
-
   return k8sCreate(PipelineModel, template, { ns: namespace });
 };
+
+export const createPipelineRunForImportFlow = async (
+  formData: GitImportFormData,
+  pipeline: Pipeline,
+): Promise<PipelineRun> => {
+  const {
+    name,
+    project: { name: namespace },
+    git,
+  } = formData;
+  const pipelineInitialValues: StartPipelineFormValues = {
+    ...convertPipelineToModalData(pipeline, true),
+    secretOpen: false,
+  };
+  pipelineInitialValues.resources = (pipelineInitialValues.resources || []).map((r) => ({
+    ...r,
+    data: {
+      ...r.data,
+      ...getResourceParams(r.data.type, r.name, name, namespace, git),
+    },
+  }));
```

Comment:
I removed the creation of resources while creating a pipeline and added it to when creating a pipelineRun since we are now automatically starting the pipeline and we need to create the pipelineRun with the required resources and this way I get the resourceRef for the resources easily added to the pipelineRun. Also I think we still need to create the required resources to successfully run the pipeline, isnt it?

Reply:
Right, that is a correct assumption to the requirement for the PipelineRun. However, PipelineResources are being moved way from in upstream and all our catalog options use Workspaces and Params to facilitate this area/gap.

So the code you removed above:
```
  if (template.spec.resources?.find((r) => r.type === 'git' && r.name === 'app-source')) {
    await createGitResource(git.url, namespace, git.ref);
  }
  if (template.spec.resources?.find((r) => r.type === 'image' && r.name === 'app-image')) {
    await createImageResource(name, namespace);
  }
```
Was for us to create the PipelineResources based off of the agreed upon names with the Operator resources.

Your `getResourceParams` only manages these said agreed upon PipelineResources. So thus it's a bit confusing removing our creation of the resources (which likely won't be used anymore) and then adding support for said resources.

I think we can just not support PipelineResources anymore in our auto start flow and if it fails to start the PipelineRun, they can go manually add a PipelineResource with the Start Modal.

Reply:
>Your getResourceParams only manages these said agreed upon PipelineResources. So thus it's a bit confusing removing our creation of the resources (which likely won't be used anymore) and then adding support for said resources.

I removed the creation from createPipelineForImportFlow and added it to createPipelineRunForImportFlow since it is required to add either the resourceSpec or resourceRef of the created resources to the PipelineRun so if I created the resources in createPipelineForImportFlow then I wouldnt know the resourceRef of the resources as the resource name is randomly generated and adding resourceSpec is also easier with submitStartPipeline as it handles everything from creating the resource to adding the correct data to the pipelineRun. 

> However, PipelineResources are being moved way from in upstream and all our catalog options use Workspaces and Params to facilitate this area/gap.

I wasnt aware of this so I will remove the creation of resources from our  our auto start flow.

Reply:
Ah, that makes sense; not the only way to solve it but definitely the most logical solution. Unfortunately, even if we had still support for PipelineResources upstream, we do not have a lot of support for inline specs anywhere in our UI. So that likely would have run into its own set of issues.